JPERF-455: Expose postproccessed performance stats

sebapawlak opened the pull request.

dagguh commented on src/main/kotlin/com/atlassian/performance/tools/report/api/judge/RelativePerformanceStabilityJudge.kt:
```diff
@@ -5,23 +5,30 @@ import com.atlassian.performance.tools.report.api.result.InteractionStats
     ): Verdict {
         val baselineDispersions = baselineStats.dispersions
         val experimentDispersions = experimentStats.dispersions
-        if (baselineDispersions == null || experimentDispersions == null) {
```
Bye bye nulls :)

dagguh commented on src/main/kotlin/com/atlassian/performance/tools/report/api/result/Stats.kt:
```diff
@@ -0,0 +1,14 @@
+import java.time.Duration
+
+/**
+ * TODO: Document this file here
```
TODO: Document this file here

dagguh replied:
https://en.wikipedia.org/wiki/Statistical_parameter :
> Parameters of this type are given names appropriate to their roles, including the following.
> * location parameter
> * dispersion parameter or scale parameter
> * shape parameter

It contains links explaining location and dispersion/scale.

dagguh commented on src/main/kotlin/com/atlassian/performance/tools/report/api/CohortsReporter.kt:
```diff
@@ -0,0 +1,60 @@
+/**
+ * Summarizes per cohort statistical results.
+ */
+class CohortsReporter(
```
"Reporter" is vague. `DistributionComparison`, `MeanLatencyChart` and `EditedIssuesReport` also report on multiple cohorts. Concretely, both `DataReporter` and `CohortsReporter` summarize the stats in a CSV table.

Some ideas to make it more concrete: `CohortsTable`, `MultiCohortStats`, `CohortsCsvStats`

sebapawlak replied:
`CohortStatsSummary`? Because it "Summarizes"

dagguh replied:
Sounds good

dagguh commented on src/main/kotlin/com/atlassian/performance/tools/report/api/DataReporter.kt:
```diff
@@ -12,6 +13,7 @@ import java.time.Duration
 /**
  * Summarizes per cohort statistical results.
  */
+@Deprecated(message = "Use CohortsReporter instead.")
```
Let's mention deprecations in the changelog

sebapawlak commented:
I think this is some sort of problem with GH because I have addressed all feedback. I looks like GH ate one of the force pushes and is not showing it in this timeline. The changes are, however, visible in the file changed tab.

dagguh commented:
Either that, or the diff doesn't work as I'd expect
